aries_vcx/tests/test_integration.rs

```diff
@@ -1001,38 +1014,104 @@ mod tests {
         let mut institution = Faber::setup();
         let mut consumer = Alice::setup();
 
+        let (consumer_to_institution, institution_to_consumer) = create_connected_connections_via_public_invite(&mut consumer, &mut institution);
+
+        institution_to_consumer.send_generic_message("Hello Alice, Faber here").unwrap();
+
+        consumer.activate().unwrap();
+        let consumer_msgs = consumer_to_institution.download_messages(Some(vec![MessageStatusCode::Received]), None).unwrap();
+        assert_eq!(consumer_msgs.len(), 1);
+    }
+
+    #[test]
+    #[cfg(feature = "agency_pool_tests")]
+    fn test_establish_connection_via_oob_message_request_included() {
+        let _setup = SetupLibraryAgencyV2::init();
+        let mut institution = Faber::setup();
+        let mut consumer = Alice::setup();
+
         institution.activate().unwrap();
-        let public_invite_json = institution.create_public_invite();
-        let public_invite: Invitation = serde_json::from_str(&public_invite_json).unwrap();
+
+        let (schema_id, _schema_json, cred_def_id, _cred_def_json, cred_def, rev_reg_id) = _create_address_schema();
+
+        let institution_did = settings::get_config_value(settings::CONFIG_INSTITUTION_DID).unwrap();
+        let _requested_attrs = requested_attrs(&institution_did, &schema_id, &cred_def_id, None, None);
+        let requested_attrs_string = serde_json::to_string(&_requested_attrs).unwrap();
+
+        let mut request_sender = create_proof_request(&mut institution, &requested_attrs_string, "[]", "{}", None);
+
+        let service = FullService::try_from(&institution.agent).unwrap();
+        let oob_sender = OutOfBand::create()
+            .set_label("test-label")
+            .set_goal_code(GoalCode::P2PMessaging)
+            .set_goal("To exchange message")
+            .append_service(ServiceResolvable::FullService(service))
+            .append_handshake_protocol(HandshakeProtocol::ConnectionV1).unwrap()
+            .append_a2a_message(request_sender.to_a2a_message()).unwrap();
+        let oob_msg = oob_sender.to_a2a_message();
 
         consumer.activate().unwrap();
-        let mut consumer_to_institution = Connection::create_with_invite("institution", public_invite, true).unwrap();
-        consumer_to_institution.connect().unwrap();
-        consumer_to_institution.update_state().unwrap();
+        let oob_receiver = OutOfBand::create_from_a2a_msg(&oob_msg).unwrap();
+        let conn = oob_receiver.connection_exists(vec![]).unwrap();
```

Comment:
there's not much point calling this if you only pass in empty vector though

Reply:
Why? It is a perfectly valid input